### PR TITLE
KAFKA-18021: Disabled MirrorCheckpointConnector throws RetriableException on task config generation

### PR DESCRIPTION
**Pull request by frankvicky**
JIRA: KAFKA-18021

When a `MirrorCheckpointConnector` is disabled, the `start` method exits early without initiating `loadInitialConsumerGroups`. 
If the connector is restarted in a disabled state, `taskConfigs` continuously throws `RetriableException` as `loadInitialConsumerGroups` never completes. 
This prevents emitting an empty set of task configs to stop running tasks, leaving previously active `MirrorCheckpointTasks` enabled and making it impossible to fully disable them.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by frankvicky** [comment]
Hi @gharris1727 
Could you please take a look when you have some time?
Many thanks 🙇🏼

File: connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java [hunk 1]
```diff
@@ -132,20 +132,23 @@ public Class<? extends Task> taskClass() {
...
         }
 
-        // if the replication is disabled, known consumer group is empty, or checkpoint emission is
-        // disabled by setting 'emit.checkpoints.enabled' to false, the interval of checkpoint emission
-        // will be negative and no 'MirrorCheckpointTask' will be created
-        if (!config.enabled() || knownConsumerGroups.isEmpty()
-                || config.emitCheckpointsInterval().isNegative()) {
+        // if known consumer group is empty, or checkpoint emission is disabled by setting 'emit.checkpoints.enabled' to false,
+        // the interval of checkpoint emission will be negative and no 'MirrorCheckpointTask' will be created
+        if (knownConsumerGroups.isEmpty() || config.emitCheckpointsInterval().isNegative()) {
```
Comment (gharris1727) [inline connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java]: Can you move the comment and the emitCheckpointsInterval check up to the earlier if-condition also?

Comment (frankvicky) [reply connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java]: Sure 😺

**Review by gharris1727 — APPROVED** [review]
LGTM, thanks @frankvicky for the fix!

**Pull request merged**